File: app/src/at-myhuynh/java/asiantech/internship/summer/java_core/ex21/SumAndPrime.java
package asiantech.internship.summer.java_core.ex21;

import java.util.ArrayList;
import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;

import asiantech.internship.summer.java_core.common.Common;

public class SumAndPrime {

    public static void main(String[] args) {
        Logger log = Logger.getLogger(SumAndPrime.class.getName());
        int n = Common.input("Nhập vào 1 số:");
        if (log.isLoggable(Level.INFO)) {
            log.info("Total: " + Common.totalOfNumber(n));
        }

        List<String> strPrime = new ArrayList<>();
        if (Common.isPrime(n)) {
            strPrime.add(String.valueOf(n));
        } else {
            strPrime = Common.primeFactor(n);
        }
        if (log.isLoggable(Level.INFO)) {
            log.info(n + " => " + Common.joinString("x", strPrime.toArray()));
        }
    }
}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/ex21/SumAndPrime.kt
package asiantech.internship.summer.kotlin.ex21

import asiantech.internship.summer.kotlin.common.Common
import asiantech.internship.summer.kotlin.common.isPrime
import asiantech.internship.summer.kotlin.common.totalOfNumber

object SumAndPrime {

    @JvmStatic
    fun main(args: Array<String>) {
        val n = Common.input("Nhập vào 1 số:")
        println("Total: " + n.totalOfNumber())

        var strPrime = mutableListOf<String>()
        if (n.isPrime()) {
            strPrime.add(n.toString())
        } else {
            strPrime = Common.primeFactor(n).toMutableList()
        }
        println("$n => ${strPrime.joinToString("x")} ")
    }
}
